### DISCUSSION TITLE
FATAL get k8s artifacts error .spec.template.spec.initContainers accessor error: <nil>

### DISCUSSION THREAD

**Discussion by viorel-anghel** [aquasecurity/trivy#5971] [Bugs]
### Description

I run 
 ` trivy k8s -n keycloak  --ignore-unfixed  --severity CRITICAL --report all   all`
and I get this error:
 ` FATAL	get k8s artifacts error: .spec.template.spec.initContainers accessor error: <nil> is of the type <nil>, expected []interface{}`
 
 When inspecting resources in that namespace, I see 
 ```
   annotations:
    kubectl.kubernetes.io/last-applied-configuration: [...] "initContainers":null, [...]
```
This is valid even if redundant but causes trivy to halt on error. 

The **workaround** was to delete with `kubectl edit` that specific part (_"initContainers":null_), after that I get a normal trivy scan.

### Desired Behavior

Ignore the annotation and run the scan normally.

### Actual Behavior

scan stopped with 

` FATAL	get k8s artifacts error: .spec.template.spec.initContainers accessor error: <nil> is of the type <nil>, expected []interface{}`

### Reproduction Steps

```bash
sorry, I no longer have the error, solved with the workaround described above.
```

### Target

Kubernetes

### Scanner

Vulnerability

### Output Format

Table

### Mode

Standalone

### Debug Output

```bash
sorry, I no longer have the error, solved with the workaround described above.
```

### Operating System

macOS Sonoma

### Version

```bash
Version: 0.48.2
Vulnerability DB:
  Version: 2
  UpdatedAt: 2024-01-19 00:16:22.405637698 +0000 UTC
  NextUpdate: 2024-01-19 06:16:22.405637327 +0000 UTC
  DownloadedAt: 2024-01-19 05:47:40.210495 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2024-01-19 00:43:43.491598305 +0000 UTC
  NextUpdate: 2024-01-22 00:43:43.491598144 +0000 UTC
  DownloadedAt: 2024-01-19 06:18:15.687657 +0000 UTC
Policy Bundle:
  Digest: sha256:f21e8e92a7b3f6042ef7acfd3b799afc1648536dc4111c4d5458bc16396f8332
  DownloadedAt: 2024-01-19 05:47:41.312449 +0000 UTC
```

### Checklist

- [X] Run `trivy image --reset`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by chen-keinan** [comment]
@viorel-anghel Closing the issue , it has beem fixed by #5939 and it will be out with `trivy 0.49.0`
